### PR TITLE
Create generator for all pieces/refactor

### PR DESCRIPTION
**Pull request by zacharybmeyers**
Remove redundant logic for validation of placing/leaving the general in check

### CODE REVIEW

File: JanggiGame.py [hunk 1]
```diff
@@ -63,6 +63,17 @@ def get_board(self):
...
+    def generate_piece_objects(self):
+        """generator yields all the piece objects from the game board"""
+        for row in self.get_board():
+            for piece_obj in row:
+                if piece_obj is not None:
+                    yield piece_obj
+
+    def get_all_pieces(self):
+        """returns a list of the current piece objects"""
+        return list(self.generate_piece_objects())
```
Comment (asiderop) [inline JanggiGame.py:75]: Keep in mind, this method will always create and return a full list. But calling the generator will return one item at a time, which could be faster overall (depending on the use-case).

File: JanggiGame.py [hunk 2]
```diff
@@ -227,32 +238,24 @@ def get_general(self, color):
...
-        for row in board:
-            for piece_obj in row:
-                # if not empty...
-                if piece_obj is not None:
-                    # if color
-                    if color in piece_obj.get_name():
-                        # get the valid moves and add them to the enemy list
-                        all_valid_moves[piece_obj.get_numeric_position()] = piece_obj.get_valid_moves()
+        for piece_obj in self.get_all_pieces():
+            if color in piece_obj.get_color():
```
Comment (asiderop) [inline JanggiGame.py:256]: Could create another generator (that extends the first) to only return one player's pieces.

File: JanggiGame.py [hunk 3]
```diff
@@ -451,18 +454,11 @@ def make_move(self, start, end):
             next_color = "b"
             next_color_for_check = "blue"
 
-        # If the move places the current player in check, invalid move
+        # At this point, the current player's move is in their valid move set, but...
+        #   If this move places or leaves the current player's general in check, invalid move
```
Comment (asiderop) [inline JanggiGame.py:458]: i.e. "ends with"

**Review by asiderop — COMMENTED** [review]
Nice refactoring!

File: JanggiGame.py (hunk 2 above)
Comment (zacharybmeyers) [reply JanggiGame.py:256]: Good idea! I'll try it out.

File: JanggiGame.py (hunk 3 above)
Comment (zacharybmeyers) [reply JanggiGame.py:458]: Thank you, much more succinct

**Pull request merged**